flink-libraries/flink-table/src/main/scala/org/apache/flink/table/runtime/aggregate/AggregateUtil.scala

```diff
@@ -1414,8 +1414,29 @@ object AggregateUtil {
           aggregates(index) = udagg.getFunction
           accTypes(index) = udagg.accType
 
-        case unSupported: SqlAggFunction =>
-          throw new TableException(s"unsupported Function: '${unSupported.getName}'")
+        case other: SqlAggFunction =>
```

Comment:
Change this case to `case collect: SqlAggFunction if collect.getKind == SqlKind.COLLECT =>` to have a dedicated case for this built-in function. Also the case after `case _: SqlCountAggFunction` to have all built-in functions together.

Reply:
done